src/remote_adapter/remote_adapter_client/remote_client/remote_client.cc

```diff
@@ -0,0 +1,112 @@
+#include "remote_client.h"
+
+#include <iostream>
+
+#include "common/constants.h"
+#include "rpc/detail/log.h"
+
+namespace lua_lib {
+
+RPCLIB_CREATE_LOG_CHANNEL(RemoteClient)
+
+RemoteClient::RemoteClient(connection_ptr connection)
+    : connection_(std::move(connection)) {
+  LOG_INFO("{}", __func__);
+  const int timeout_ = 10000;
+  connection_->set_timeout(timeout_);
+
+  LOG_INFO("Check connection: ");
+
+  auto response =
+      connection_->call(constants::client_connected).as<response_type>();
+
+  if (constants::error_codes::SUCCESS != response.second) {
+    LOG_ERROR("{}", response.first);
+  } else {
+    LOG_INFO("connection OK");
+  }
+}
+
+bool RemoteClient::connected() const {
+  LOG_INFO("{} Check connection:", __func__);
+  if (rpc::client::connection_state::connected ==
+      connection_->get_connection_state()) {
+    LOG_INFO("connection OK");
+    return true;
+  }
+  LOG_ERROR("Not connected");
+  return false;
+}
+
+response_type RemoteClient::file_call(const std::string &rpc_name,
+                                      std::vector<parameter_type> parameters) {
+  LOG_INFO("{0}: of: {1} with {2} parameters", __func__, rpc_name,
+           parameters.size());
+
+  if (connected()) {
+    parameters.push_back(
+        std::make_pair(std::to_string(0), constants::param_types::INT));
+    parameters.push_back(std::make_pair(std::to_string(constants::kMaxSizeData),
+                                        constants::param_types::INT));
+    response_type response =
+        connection_->call(rpc_name, parameters).as<response_type>();
+
+    if (constants::error_codes::FAILED == response.second) {
+      LOG_ERROR("{0}:\nExit Get file data from HU Failed!!!", __func__);
+      return response;
+    }
+
+    std::string tmp_path(constants::kTmpPath);
+    tmp_path.append(parameters[1].first);
+
+    remove(tmp_path.c_str());
```

Comment:
Where is this call defined? I can't find it in the project

Reply:
@jacobkeeler 
This call defined in stdio.h - Remove file